### PR TITLE
Issue #3383: Add LDAP registry JNDI read timeout.

### LINKED ISSUE #3383: ldapRegistry-3.0 does not configure a read timeout for JNDI connections
The ldapRegistry-3.0 feature does not allow the user to configure, nor does it configure implicitly a read timeout on the JNDI context. This can lead to hangs when an LDAP server does not respond.

Set the read timeout either based on the connect timeout that is already exposed or add a separate read timeout. This value should align closely with the "searchTimeout" used on the SearchControls as the time limit for a search operation.

### PR DESCRIPTION
**Pull request by jvanhill**
fixes #3383

### CODE REVIEW

**Comment by jvanhill** [comment]
#build

**Comment by jvanhill** [comment]
#build

File: dev/com.ibm.ws.security.wim.adapter.ldap/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -60,10 +60,13 @@ recursiveSearch=Perform a nested group search
...
-searchTimeout.desc=Maximum time for an LDAP server to respond before a request is canceled. This is equivalent to a read timeout once the connection is established.
+searchTimeout.desc=Maximum time for an LDAP server to respond before a request is canceled.
 
-connectTimeout=Timeout duration for establishing LDAP connection
-connectTimeout.desc=Maximum time for establishing a connection to the LDAP server. The program logs an error message if the specified time expires.
+connectTimeout=Timeout for establishing LDAP connection
+connectTimeout.desc=Maximum time for establishing a connection to the LDAP server. A value of 0 indicates that the TCP protocol's timeout value should be used. The program logs an error message if the specified time expires.
+
+readTimeout=Timeout for reading from an LDAP connection
+readTimeout.desc=Maximum time to wait for read operations for LDAP operations. A value of 0 indicates that there is no timeout and the read will wait indefinitely. The program logs an error message if the specified time expires.
```
Comment (chirp1) [inline dev/com.ibm.ws.security.wim.adapter.ldap/resources/OSGI-INF/l10n/metatype.properties]: Change "A value of 0 indicates that there is no timeout and the read will wait indefinitely." to "A value of 0 indicates that no timeout exists and that the read waits indefinitely."

File: dev/com.ibm.ws.security.wim.adapter.ldap/resources/OSGI-INF/l10n/metatype.properties [hunk 2]
```diff
@@ -60,10 +60,13 @@ recursiveSearch=Perform a nested group search
 recursiveSearch.desc=Performs a nested group search. Select this option only if the LDAP server does not support recursive server-side searches.
 
 searchTimeout=Search timeout
-searchTimeout.desc=Maximum time for an LDAP server to respond before a request is canceled. This is equivalent to a read timeout once the connection is established.
+searchTimeout.desc=Maximum time for an LDAP server to respond before a request is canceled.
 
-connectTimeout=Timeout duration for establishing LDAP connection
-connectTimeout.desc=Maximum time for establishing a connection to the LDAP server. The program logs an error message if the specified time expires.
+connectTimeout=Timeout for establishing LDAP connection
```
Comment (chirp1) [inline dev/com.ibm.ws.security.wim.adapter.ldap/resources/OSGI-INF/l10n/metatype.properties]: Change "Timeout for establishing LDAP connection" to "Timeout for establishing an LDAP connection". (Add "an" and add a period at the end.)

File: dev/com.ibm.ws.security.wim.adapter.ldap/resources/OSGI-INF/l10n/metatype.properties (hunk 1 above)
Comment (jvanhill) [reply dev/com.ibm.ws.security.wim.adapter.ldap/resources/OSGI-INF/l10n/metatype.properties]: I have updated the message as suggested.

File: dev/com.ibm.ws.security.wim.adapter.ldap/resources/OSGI-INF/l10n/metatype.properties (hunk 2 above)
Comment (jvanhill) [reply dev/com.ibm.ws.security.wim.adapter.ldap/resources/OSGI-INF/l10n/metatype.properties]: I have updated the label wording, but did not add the period as labels should not be sentences. I believe the build fail if I do.

**Comment by jvanhill** [comment]
#build

**Pull request merged**